docs/src/pages/customization/how-to-customize/how-to-customize.md

```diff
@@ -145,8 +145,31 @@ Here are four alternatives; each has its pros and cons.
 
 ### Dynamic CSS
 
+Using the `styled()` utility offers a simple way for adding dynamic styles based on props.
+
 {{"demo": "pages/customization/how-to-customize/DynamicCSS.js", "defaultCodeOpen": false}}
 
+> ⚠️ Note that if you are using TypeScript you will need to update the prop's types of the new component.
+
+```tsx
+import * as React from 'react';
+import { experimentalStyled as styled } from '@material-ui/core/styles';
+import Slider, { SliderProps } from '@material-ui/core/Slider';
+
+type StyledSliderProps = SliderProps & {
+  success?: boolean;
+};
+
+const StyledSlider = styled(Slider, {
+  shouldForwardProp: (prop) => prop !== 'success',
+})<StyledSliderProps>(({ success, theme }) => ({
```

Comment:
question: Do we need a generic or does 
```suggestion
})(({ success, theme }: StyledSliderProps) => ({
```

work as well? The suggested code is more "vanilla" JavaScript (i.e. less TypeScript syntax) and allows extracting the function.

Reply:
This should work, let me check again and update if that's the case

Reply:
Actually, the `theme` prop is added on top of the generic used in the example. the `StyledSliderProps` themself do not contain that prop. That's why the generic is required.

Reply:
Alright, the generic it is then.